client/allocdir/task_dir.go

```diff
@@ -8,80 +8,105 @@ import (
 	"os"
 	"path/filepath"
 
-	hclog "github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/go-set/v2"
+	"github.com/hashicorp/nomad/helper/users/dynamic"
+	"github.com/hashicorp/nomad/plugins/drivers/fsisolation"
 )
 
 // TaskDir contains all of the paths relevant to a task. All paths are on the
 // host system so drivers should mount/link into task containers as necessary.
 type TaskDir struct {
-	// AllocDir is the path to the alloc directory on the host
+	// AllocDir is the path to the alloc directory on the host.
+	// (not to be conflated with client.alloc_dir)
+	//
+	// <alloc_dir>
 	AllocDir string
 
-	// Dir is the path to Task directory on the host
+	// Dir is the path to Task directory on the host.
+	//
+	// <task_dir>
 	Dir string
 
+	// MountsAllocDir is the path to the alloc directory on the host that has
+	// been bind mounted under <client.mounts_dir>
+	//
+	// <client.mounts_dir>/task/<alloc_dir> -> <alloc_dir>
```

Comment:
I don't think the documented source path here matches what we ended up doing with the unique task name directory? (Same for `MountsTaskDir` below.)

Reply:
ooh yeah that was from before I figured out how things would look, fixed in 4a45eaf355